tests/rkt_api_service_bench_test.go

```diff
@@ -0,0 +1,195 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/golang.org/x/net/context"
+	"github.com/coreos/rkt/api/v1alpha"
+	"github.com/coreos/rkt/tests/testutils"
+)
+
+func benchPods(b *testing.B, numOfPods int, detail, benchListPods, benchInspectPod bool) {
+	t := new(testing.T)
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	svc := startAPIService(t, ctx)
+	defer stopAPIService(t, svc)
+
+	c, conn := newAPIClientOrFail(t, "localhost:15441")
+	defer conn.Close()
+
+	imagePath := patchTestACI("rkt-inspect-print.aci", "--exec=/inspect --print-msg=HELLO_API --exit-code=42")
+	defer os.Remove(imagePath)
+
+	cmd := fmt.Sprintf("%s --insecure-skip-verify run %s", ctx.Cmd(), imagePath)
+
+	var wg sync.WaitGroup
+	wg.Add(numOfPods)
+	for i := 0; i < numOfPods; i++ {
+		go func() {
+			spawnAndWaitOrFail(t, cmd, true)
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+
+	if benchListPods {
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			_, err := c.ListPods(context.Background(), &v1alpha.ListPodsRequest{Detail: detail})
+			if err != nil {
+				b.Error(err)
+			}
+		}
+		b.StopTimer()
+		return
+	}
+
+	if benchInspectPod {
+		resp, err := c.ListPods(context.Background(), &v1alpha.ListPodsRequest{})
+		if err != nil {
+			b.Fatalf("Unexpected error: %v", err)
+		}
+
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			_, err := c.InspectPod(context.Background(), &v1alpha.InspectPodRequest{Id: resp.Pods[0].Id})
+			if err != nil {
+				b.Error(err)
+			}
+		}
+		b.StopTimer()
+	}
+}
+
+func benchImages(b *testing.B, numOfImages int, detail, benchListImages, benchInspectImage bool) {
+	t := new(testing.T)
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	svc := startAPIService(t, ctx)
+	defer stopAPIService(t, svc)
+
+	c, conn := newAPIClientOrFail(t, "localhost:15441")
+	defer conn.Close()
+
+	var wg sync.WaitGroup
+	wg.Add(numOfImages)
+	for i := 0; i < numOfImages; i++ {
+		go func(i int) {
+			patchImportAndFetchHash(fmt.Sprintf("rkt-inspect-sleep-%d.aci", i), []string{"--exec=/inspect"}, t, ctx)
+			wg.Done()
+		}(i)
+	}
+	wg.Wait()
+
+	if benchListImages {
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			_, err := c.ListImages(context.Background(), &v1alpha.ListImagesRequest{Detail: true})
+			if err != nil {
+				b.Error(err)
+			}
+		}
+		b.StopTimer()
+		return
+	}
+
+	if benchInspectImage {
+		resp, err := c.ListImages(context.Background(), &v1alpha.ListImagesRequest{})
+		if err != nil {
+			b.Fatalf("Unexpected error: %v", err)
+		}
+
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			_, err := c.InspectImage(context.Background(), &v1alpha.InspectImageRequest{Id: resp.Images[0].Id})
+			if err != nil {
+				b.Error(err)
+			}
+		}
+		b.StopTimer()
+	}
+}
+
+func BenchmarkList1PodNoDetail(b *testing.B) {
+	benchPods(b, 1, false, true, false)
```

Comment:
I know this is just test/benchmark code, but benchPods() use of these bools makes for pretty unreadable code.